### ISSUE TITLE
AuthorizationManagementClient broken

### ISSUE THREAD

**Issue by cshipman** [Azure/azure-sdk-for-node#3638]
I'm trying to automate the provisioning of resources in Azure and I have stumbled on adding principles as owners to a resource group. When I try to use the authorization management client I get the below error (this happens for roleDefinitions as well). 

```javascript
Error: The resource type 'roleAssignments' could not be found in the namespace 'Microsoft.Authorization' for api version '2018-09-01-preview'. The supported api-versions are '2014-04-01-preview,2014-07-01-preview,2014-10-01-preview,2015-05-01-preview,2015-06-01,2015-07-01,2016-07-01,2017-05-01,2017-09-01,2017-10-01-preview,2018-01-01-preview'
```

I can't even work around it because you hardcoded the apiVersion and provide no way for me to specify an older one...

**Comment by RikkiGibson** [comment]
We released `azure-arm-authorization@8.0.0` which no longer makes a reference to API version 2018-09-01-preview. Please follow up if you have any further questions or discussion about this issue.

**Comment by alexjchapman** [comment]
I'm using `azure-arm-authorization@8.0.0` but I'm getting the following error:

```Error: The api-version '2018-09-01-preview' is invalid. The supported versions are '2018-11-01,2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'``` 

Again it looks like the api-version has been hardcoded.
